src/loader/XSensRecordingReader.py

```diff
@@ -0,0 +1,153 @@
+import os
+import pandas as pd
+import re
+import json
+import utils.settings as settings
+
+verbose = False
+
+if verbose:
+
+    def d_print(*args):
+        print(*args)
+
+else:
+
+    def d_print(*args):
+        pass
+
+
+class XSensRecordingReader(object):
+    """
+    XSensRecordingReader.get_recording_frame() concatenates all sensor csv files in the recording folder to one data frame
+    """
+
+    @staticmethod
+    def get_recording_frame(recording_folder_path) -> pd.DataFrame:
+        """
+        All files in format 'ID_timestamp.csv' should be in recording_folder_path to get loaded
+        """
+
+        # This is the output variable
+        recording_frame = None
+        # Iterate over all files in the data directory
+        for file_name in os.listdir(recording_folder_path):
+            is_file = os.path.isfile(os.path.join(recording_folder_path, file_name))
+            if not is_file or not file_name.endswith(".csv"):
+                continue
+            sensor_file_name = file_name
+
+            # Extract the sensor ID from the filename
+            mac_regex = re.compile(r"(?:[0-9a-fA-F]-?){12}", re.IGNORECASE)
+            sensor_mac_address = re.findall(mac_regex, sensor_file_name)[0]
+            with open(recording_folder_path + os.path.sep + "metadata.json", "r") as f:
+                data = json.load(f)
+            sensor_map_suffix_map = data["sensorMapping"]
+
+            # map from mac address to placements (LW, RW, etc.) without sensor set information
```

Comment:
maybe we should have already done this, when saving the data? this will be hard for others to understand when working with our dataset

Reply:
I'm not totally sure what you mean, but I also didnt completely write this, but copy it from our old repo. Maybe we can talk about it tomorrow